account_invoice_report_grouped_by_picking_hide_pickings/models/account_move.py

```diff
@@ -1,15 +1,56 @@
 # Copyright 2024 Alberto Martínez <alberto.martinez@sygel.es>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import models
+from odoo import api, models
 
 
 class AccountMove(models.Model):
     _inherit = "account.move"
 
+    @api.model
+    def _update_lines_grouped_by_picking(self, lines, removed):
+        """Updates the invoice lines after removing some of them
+        to keep the total quantity consistent,
+        by creating correction lines
+        or modifying old correction lines's quantity
+        """
+        # Store the removed qty
+        to_update = {}
+        for line in removed:
+            key = line["line"]
+            if not to_update.get(key):
+                to_update[key] = line["quantity"]
+            elif to_update[key] + line["quantity"] != 0:
+                to_update[key] += line["quantity"]
+            else:
+                to_update.pop(key)
+        # Update the remaining qty
+        for inv_line, qty in to_update.items():
+            line_without_picking = list(
+                filter(lambda g: not g["picking"] and g["line"] == inv_line, lines)
+            )[:1]
+            if not line_without_picking:
+                lines += {
+                    "picking": self.env["stock.picking"],
```

Comment:
indicating the model objetc and a false would not be the same?

Reply:
It would be the same, but the orginal module does not set the picking as False, it sets it as the empty recordset. I prefer to keep this similar, maybe another module depends of this module and assumes this is not False